code/game/sound/sound.dm

```diff
@@ -0,0 +1,224 @@
+			return //No sound
+
+		var/dx = turf_source.x - turf_loc.x // Hearing from the right/left
+		sound_to_use.x = dx * distance_multiplier
+		var/dz = turf_source.y - turf_loc.y // Hearing from infront/behind
+		sound_to_use.z = dz * distance_multiplier
+		var/dy = (turf_source.z - turf_loc.z) * 5 * distance_multiplier // Hearing from  above / below, multiplied by 5 because we assume height is further along coords.
+		sound_to_use.y = dy
+
+		sound_to_use.falloff = max_distance || 1 //use max_distance, else just use 1 as we are a direct sound so falloff isnt relevant.
+
+		// Sounds can't have their own environment. A sound's environment will be:
+		// 1. the mob's
+		// 2. the area's (defaults to SOUND_ENVRIONMENT_NONE)
+		if(sound_environment_override != SOUND_ENVIRONMENT_NONE)
+			sound_to_use.environment = sound_environment_override
+		else
+			var/area/A = get_area(src)
+			sound_to_use.environment = A.sound_environment
+
+		if(use_reverb && sound_to_use.environment != SOUND_ENVIRONMENT_NONE) //We have reverb, reset our echo setting
+			sound_to_use.echo[3] = 0 //Room setting, 0 means normal reverb
+			sound_to_use.echo[4] = 0 //RoomHF setting, 0 means normal reverb.
+
+	SEND_SOUND(src, sound_to_use)
+
+/proc/sound_to_playing_players(soundin, volume = 100, vary = FALSE, frequency = 0, channel = 0, pressure_affected = FALSE, sound/S)
+	if(!S)
+		S = sound(get_sfx(soundin))
+	for(var/m in GLOB.player_list)
+		if(ismob(m) && !isnewplayer(m))
+			var/mob/M = m
+			M.playsound_local(M, null, volume, vary, frequency, null, channel, pressure_affected, S)
+
+/mob/proc/stop_sound_channel(chan)
+	SEND_SOUND(src, sound(null, repeat = 0, wait = 0, channel = chan))
+
+/mob/proc/set_sound_channel_volume(channel, volume)
+	var/sound/S = sound(null, FALSE, FALSE, channel, volume)
+	S.status = SOUND_UPDATE
+	SEND_SOUND(src, S)
+
+/client/proc/playtitlemusic(vol = 85)
+	set waitfor = FALSE
+	UNTIL(SSticker.login_music) //wait for SSticker init to set the login music
+
+	if(prefs && (prefs.read_preference(/datum/preference/toggle/sound_lobby)) && !CONFIG_GET(flag/disallow_title_music))
+		SEND_SOUND(src, sound(SSticker.login_music, repeat = 0, wait = 0, volume = vol, channel = CHANNEL_LOBBYMUSIC)) // MAD JAMS
+
+/proc/get_rand_frequency()
+	return rand(32000, 55000) //Frequency stuff only works with 45kbps oggs.
+
+///Used to convert a SFX define into a .ogg so we can add some variance to sounds. If soundin is already a .ogg, we simply return it
+/proc/get_sfx(soundin)
+	if(!istext(soundin))
+		return soundin
+
+	var/static/list/sfx_by_id
+	if(isnull(sfx_by_id))
+		sfx_by_id = list()
+		for(var/sfx_subtype in subtypesof(/datum/sfx))
+			var/datum/sfx/sfx = new sfx_subtype()
+			if(isnull(sfx.id))
+				stack_trace("SFX category [sfx_subtype]` has no `id` defined")
+				continue
+
+			if(length(sfx.sound_files) == 0)
+				stack_trace("SFX category [sfx_subtype] has no sound files associated")
+				continue
+
+			sfx_by_id[sfx.id] = sfx
```

Comment:
I think this would be better off as a global list. At the very least, you'd be able to split up the initialization of the list from the get_sfx proc, which should ideally be simple. Additionally, any runtimes would be caught upon initialization vs whenever a sound is played. I don't think the increase in init cost would be significant enough to not do this.

Reply:
Ye, why not. The reason behind the static list was that I wanted to keep it in this proc.